hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/AbstractRpcClient.java

```diff
@@ -527,6 +528,21 @@ public RpcChannel createRpcChannel(ServerName sn, User user, int rpcTimeout)
     return new RpcChannelImplementation(this, createAddr(sn), user, rpcTimeout);
   }
 
+  @Override
+  public RpcChannel createHedgedRpcChannel(Set<ServerName> sns, User user, int rpcTimeout)
+      throws UnknownHostException {
+    final int hedgedRpcFanOut = conf.getInt(HConstants.HBASE_RPCS_HEDGED_REQS_FANOUT_KEY,
+        HConstants.HBASE_RPCS_HEDGED_REQS_FANOUT_DEFAULT);
+    Set<InetSocketAddress> addresses = new HashSet<>();
+    for (ServerName sn: sns) {
+      addresses.add(createAddr(sn));
+    }
+    Preconditions.checkState(this instanceof NettyRpcClient,
+        "Hedging only supported for non-blocking connection implementations.");
```

Comment:
Might as well use the inheritance language feature as intended. The abstract parent can provide no implementation at all, or just throw the exception. The implementation of `NettyRpcClient` can override that implementation with its own.